daiquiri/datalink/adapter.py

```diff
@@ -56,14 +64,38 @@ def get_links(self, resource_type, resource):
         return getattr(self, 'get_%s_links' % resource_type)(resource)
 
     def get_context_data(self, request, **kwargs):
+        '''Get the datalink related context data for a given request
+        '''
         context = {}
 
         if 'ID' in kwargs:
-            context['datalinks'] = Datalink.objects.filter(ID=kwargs['ID']).order_by('semantics')
+            field_names = [field['name'] for field in DATALINK_FIELDS]
+            # more precise would be to use a serializer instead of list(QuerySet.values())
+            context['datalinks'] = list(Datalink.objects.filter(ID=kwargs['ID']).order_by('semantics').values())
             context['ID'] = kwargs['ID']
 
         return context
 
+    def get_datalink_rows(self, identifiers, **kwargs):
+        '''Get the list of datalink entries for the provided identifiers
+        '''
+
+        # get the datalink entries from Datalink Table and metadata (Table and Schema)
+        field_names = [field['name'] for field in DATALINK_FIELDS]
+        rows = list(Datalink.objects.filter(ID__in=identifiers).values_list(*field_names))
+
+        # get the dynamic datalink entries
+        dyn_rows = [tuple(link.values()) for link in self.get_dyn_datalink_links(identifiers)]
```

Comment:
`tuple(link.values())` does not guaranty the order of the items in the resulting array.